valhalla/baldr/traffictile.h

```diff
@@ -62,10 +62,18 @@ inline uint16_t valhalla_traffic_seconds_to_age_bucket(const int seconds) {
 }
 
 struct Speed {
-  uint16_t speed_kmh : 8;        // km/h - so max range is 0-255km/h
-  uint16_t congestion_level : 3; // 0 - unknown, 1-6 - low-high, 7 - unused
-  uint16_t age_bucket : 4;       // Age bucket for the speed record (see SPEED_AGE_BUCKET_SIZE)
-  uint16_t spare : 1;            // TODO: reserved for later use
+  uint32_t speed1_kmh : 8; // 0-255km/h
+  uint32_t speed2_kmh : 8;
+  uint32_t speed3_kmh : 8;
+  uint32_t breakpoint1 : 8; // position = length * breakpoint1 * 255
```

Comment:
is the comment correct here? would position be in meters along the edge? the idea is that we have a resolution of 256 bins along an edge at which to place a "marker" right? if so i was thinking the comment should be:

    double meters_along1 = edge.length * breakpoint1 * (1.0/255.0);
    double meters_along2 = edge.length * breakpoint2 * (1.0/255.0);

i think for the sake of making getting the speed of an edge as fast as possible we should have a quick way to compute that linear combination:

```c++
constexpr double scale = 1.0/255.0;
double speed = speed1_kmh * breakpoint1 * scale + speed2_kmh * (breakpoint2-breakpoint1) * scale + speed3_kmh * (1 - breakpoint2 * scale);
```

importantly this means we need to zero out the speeds for segments that dont exist so that their terms go to 0 as well.

Reply:
we also need to consider when we only have partial coverage of the edge. it would be great to have a formula here that doesnt have branches in it.